src/components/App/App.js

```diff
@@ -82,6 +99,64 @@ function App() {
       });
   };
 
+  const handleLogout = () => {
+    setCurrentUser({});
+    localStorage.removeItem("jwt");
+    localStorage.removeItem("searchResults");
+    setLoggedIn(false);
+    history.push("/");
+  };
+
+  const handleBookmarkClick = (selectedCard) => {
+    // Map and create a new search results array
+    const newCards = searchResults.map((card) => {
+      return {
+        ...card,
+        isBookmarked:
+          card.key === selectedCard.key
+            ? !card.isBookmarked
+            : card.isBookmarked,
+      };
+    });
+    // either add or delete selectedCard based on selectedCard.isBookmarked field
+    if (selectedCard.isBookmarked) {
+      // delete the card in db
+      api.removeCardBookmark(selectedCard);
+      selectedCard.isBookmarked = false;
+      api.getSavedArticles().then((response) => {
+        setSavedArticles(response);
+      });
+    } else {
+      // add the card to db
+      api.addCardBookmark(selectedCard);
+      api.getSavedArticles().then((response) => {
+        setSavedArticles(response);
+      });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**